crates/superposition_types/src/database/models/experimentation.rs

```diff
@@ -288,17 +288,93 @@ pub struct EventLog {
 #[cfg_attr(feature = "diesel_derives", diesel(check_for_backend(diesel::pg::Pg)))]
 #[cfg_attr(feature = "diesel_derives", diesel(primary_key(experiment_group_id)))]
 pub struct ExperimentGroup {
-    pub experiment_group_id: String,
+    #[serde(with = "i64_formatter")]
+    pub id: i64,
+    pub context_hash: String,
     pub name: String,
     pub description: Description,
     pub change_reason: ChangeReason,
     pub context: Condition,
     pub traffic_percentage: TrafficPercentage,
-    pub member_experiment_ids: Vec<String>,
+    #[serde(with = "i64_vec_formatter")]
+    pub member_experiment_ids: Vec<i64>,
     pub created_at: DateTime<Utc>,
     pub created_by: String,
     pub last_modified_at: DateTime<Utc>,
     pub last_modified_by: String,
 }
 
 pub type ExperimentGroups = Vec<ExperimentGroup>;
+
+pub mod i64_formatter {
+    use serde::{self, Deserialize, Deserializer, Serializer};
+
+    // Serialize i64 to String
+    pub fn serialize<S>(value: &i64, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&value.to_string())
+    }
+
+    // Deserialize String to i64
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<i64, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        s.parse::<i64>()
+            .map_err(|e| serde::de::Error::custom(format!("Failed to parse i64: {}", e)))
+    }
```

Comment:
this deserialisation exclusively works for string input only, this should have been an either or to achieve [this](https://github.com/juspay/superposition/pull/525/files#r2113409900)

```suggestion
    pub fn deserialize<'de, D>(deserializer: D) -> Result<i64, D::Error>
    where
        D: Deserializer<'de>,
    {
        let value = Value::deserialize(deserializer)?;
        match value {
        Value::String(s) => s.parse::<i64>()
                    .map_err(|e| serde::de::Error::custom(format!("Failed to parse to i64: {}", e)))
        }
        Value::Number(n) => {
            if let Some(id) = n.as_i64() {
                Ok(id)
            } else {
                log::error!("Expected a bigint as input");
                Err(serde::de::Error::custom(
                    "Expected a bigint as input",
                ))
            }
        }
      _ => {
                log::error!("Expected a bigint or stringified bigint as input");
                Err(serde::de::Error::custom(
                    "Expected a bigint or stringified bigint as input",
                ))
      }
    }
```

Reply:
This a converter though, it takes a string and makes it an int. If you're passing int, why bother?